controllers/mutation_reconcile_looper.go

```diff
@@ -91,142 +97,60 @@ func (m *MutationReconcileLooper) ReconcileMutationObject(ctx context.Context, s
-	}
-
-	if err := ocmruntime.DefaultYAMLEncoding.Unmarshal(content, config); err != nil {
-		return "",
-			fmt.Errorf("failed to unmarshal content: %w", err)
-	}
+	var (
+		identity  v1alpha1.Identity
+		sourceDir string
+	)
 
-	log.Info("preparing localization substitutions")
+	if spec.ConfigRef != nil {
+		var rules localize.Substitutions
 
-	componentDescriptor, err := component.GetComponentDescriptor(ctx, m.Client, spec.ConfigRef.Resource.ResourceRef.ReferencePath, componentVersion.Status.ComponentDescriptor)
-	if err != nil {
-		return "", fmt.Errorf("failed to get component descriptor from version")
-	}
-	if componentDescriptor == nil {
-		return "", fmt.Errorf("couldn't find component descriptor for reference '%s' or any root components", spec.ConfigRef.Resource.ResourceRef.ReferencePath)
-	}
-
-	var rules localize.Substitutions
-	if spec.Values != nil {
-		rules, err = m.createSubstitutionRulesForConfigurationValues(spec, *config)
-		if err != nil {
-			return "", fmt.Errorf("failed to create configuration values for config '%s': %w", config.Name, err)
-		}
-	} else {
-		rules, err = m.createSubstitutionRulesForLocalization(ctx, *componentDescriptor, *config)
+		virtualFS, err := osfs.NewTempFileSystem()
 		if err != nil {
-			return "", fmt.Errorf("failed to create localization values for config '%s': %w", config.Name, err)
+			return "", fmt.Errorf("fs error: %w", err)
 		}
-	}
-
-	if len(rules) == 0 {
-		log.Info("no rules generated from the available config data; the generate snapshot will have no modifications")
-	}
-
-	virtualFS, err := osfs.NewTempFileSystem()
-	if err != nil {
-		return "", fmt.Errorf("fs error: %w", err)
-	}
-	defer vfs.Cleanup(virtualFS)
-
-	if err := utils.ExtractTarToFs(virtualFS, bytes.NewBuffer(resourceData)); err != nil {
-		return "", fmt.Errorf("extract tar error: %w", err)
-	}
-
-	if err := localize.Substitute(rules, virtualFS); err != nil {
-		return "", fmt.Errorf("localization substitution failed: %w", err)
-	}
-
-	fi, err := virtualFS.Stat("/")
-	if err != nil {
-		return "", fmt.Errorf("fs error: %w", err)
-	}
+		defer vfs.Cleanup(virtualFS)
 
-	sourceDir := filepath.Join(os.TempDir(), fi.Name())
+		if err := utils.ExtractTarToFs(virtualFS, bytes.NewBuffer(resourceData)); err != nil {
+			return "", fmt.Errorf("extract tar error: %w", err)
+		}
 
-	artifactPath, err := os.CreateTemp("", "snapshot-artifact-*.tgz")
-	if err != nil {
-		return "", fmt.Errorf("fs error: %w", err)
-	}
-	defer os.Remove(artifactPath.Name())
+		fi, err := virtualFS.Stat("/")
+		if err != nil {
+			return "", fmt.Errorf("fs error: %w", err)
+		}
 
-	if err := BuildTar(artifactPath.Name(), sourceDir); err != nil {
-		return "", fmt.Errorf("build tar error: %w", err)
-	}
+		sourceDir = filepath.Join(os.TempDir(), fi.Name())
 
-	version := resourceRef.Version
-	if version == "" {
-		version = "latest"
-	}
+		rules, identity, err = m.generateSubstRules(ctx, componentVersion, spec, resourceData)
```

Comment:
Hm, something is not okay here. The `resourceData` from here is actually not used at all in `generateSubstRules` function.

Reply:
You're right we don't need it. The resource data is extracted later and not within this function.